### ISSUE TITLE
example dataset start_schema cannot make sucessfully

### ISSUE THREAD

**Issue by hongbinzuo** [yandex/ClickHouse#24]
Thank you for open sourcing the great project!

Today I want to use the free dataset of [star_schema](https://github.com/yandex/ClickHouse/blob/master/doc/example_datasets/star_schema.txt), but cannot compile successfully, the error message is shown as follows,

```
root@f90500858fe7:/work/datasets/ssb-dbgen# make
gcc -O -DDBNAME=\"dss\" -DLinux -DDB2  -DSSBM    -c -o driver.o driver.c
In file included from driver.c:55:0:
dss.h:201:6: error: conflicting types for 'getopt'
 int  getopt PROTO((int arg_cnt, char **arg_vect, char *oprions));
      ^
In file included from /usr/include/unistd.h:874:0,
                 from driver.c:12:
/usr/include/getopt.h:150:12: note: previous declaration of 'getopt' was here
 extern int getopt (int ___argc, char *const *___argv, const char *__shortopts)
            ^
driver.c: In function 'partial':
driver.c:605:20: warning: format '%d' expects argument of type 'int', but argument 4 has type 'long int' [-Wformat=]
   fprintf (stderr, "\tStarting to load stage %d of %d for %s...",
                    ^
driver.c: In function 'pload':
driver.c:634:20: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
   fprintf (stderr, "Starting %d children to load %s",
                    ^
driver.c: In function 'main':
driver.c:1019:5: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
     "Generating update pair #%d for %s [pid: %d]",
     ^
driver.c:1118:24: warning: format '%ld' expects argument of type 'long int', but argument 5 has type '__pid_t {aka int}' [-Wformat=]
       fprintf (stderr, "%s data for %s [pid: %ld]",
                        ^
driver.c:1125:13: warning: format '%d' expects argument of type 'int', but argument 3 has type 'long int' [-Wformat=]
      printf("Validation checksum for %s at %d GB: %0x\n",
             ^
driver.c:1125:13: warning: format '%x' expects argument of type 'unsigned int', but argument 4 has type 'long unsigned int' [-Wformat=]
<builtin>: recipe for target 'driver.o' failed
make: *** [driver.o] Error 1
```

My env:

```
root@f90500858fe7:/work/datasets/ssb-dbgen# cat /etc/lsb-release
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04 LTS"

root@f90500858fe7:/work/datasets/ssb-dbgen# env
HOSTNAME=f90500858fe7
TERM=xterm
LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
DISABLE_MONGODB=1
PWD=/work/datasets/ssb-dbgen
THREADS=1
CXX=g++-5
SHLVL=1
HOME=/root
CC=gcc-5
_=/usr/bin/env
OLDPWD=/work/datasets
```

Can you help on this?

Thanks,
Hongbin

**Comment by alexey-milovidov** [comment]
Example to load this dataset was unfinished, I was not loaded it.
I appreciate if someone going to help.

**Comment by alexey-milovidov** [comment]
In the meantime `ssb-dbgen` tool has been fixed and now it compiles with gcc 7.3 without issues.
Please note that `ssb-dbgen` is very old fashioned third-party tool and there are numerous warnings during build, but it works fine.

Benchmark setup description can be found here:
https://www.altinity.com/blog/2017/6/16/clickhouse-in-a-general-analytical-workload-based-on-star-schema-benchmark

It's also a bit outdated, because modern ClickHouse features (custom partitioning, LowCardinality types) are missing.

You can load data with the following instruction:

```
git clone git@github.com:vadimtk/ssb-dbgen.git
cd ssb-dbgen
make

./dbgen -s 1000 -T c
./dbgen -s 1000 -T l
./dbgen -s 1000 -T p
./dbgen -s 1000 -T s
./dbgen -s 1000 -T d


CREATE TABLE customer
(
        C_CUSTKEY       UInt32,
        C_NAME          String,
        C_ADDRESS       String,
        C_CITY          LowCardinality(String),
        C_NATION        LowCardinality(String),
        C_REGION        LowCardinality(String),
        C_PHONE         String,
        C_MKTSEGMENT    LowCardinality(String)
)
ENGINE = MergeTree ORDER BY (C_CUSTKEY);

CREATE TABLE lineorder
(
    LO_ORDERKEY             UInt32,
    LO_LINENUMBER           UInt8,
    LO_CUSTKEY              UInt32,
    LO_PARTKEY              UInt32,
    LO_SUPPKEY              UInt32,
    LO_ORDERDATE            Date,
    LO_ORDERPRIORITY        LowCardinality(String),
    LO_SHIPPRIORITY         UInt8,
    LO_QUANTITY             UInt8,
    LO_EXTENDEDPRICE        UInt32,
    LO_ORDTOTALPRICE        UInt32,
    LO_DISCOUNT             UInt8,
    LO_REVENUE              UInt32,
    LO_SUPPLYCOST           UInt32,
    LO_TAX                  UInt8,
    LO_COMMITDATE           Date,
    LO_SHIPMODE             LowCardinality(String)
)
ENGINE = MergeTree PARTITION BY toYear(LO_ORDERDATE) ORDER BY (LO_ORDERDATE, LO_ORDERKEY);

CREATE TABLE part
(
        P_PARTKEY       UInt32,
        P_NAME          String,
        P_MFGR          LowCardinality(String),
        P_CATEGORY      LowCardinality(String),
        P_BRAND         LowCardinality(String),
        P_COLOR         LowCardinality(String),
        P_TYPE          LowCardinality(String),
        P_SIZE          UInt8,
        P_CONTAINER     LowCardinality(String)
)
ENGINE = MergeTree ORDER BY P_PARTKEY;

CREATE TABLE supplier
(
        S_SUPPKEY       UInt32,
        S_NAME          String,
        S_ADDRESS       String,
        S_CITY          LowCardinality(String),
        S_NATION        LowCardinality(String),
        S_REGION        LowCardinality(String),
        S_PHONE         String
)
ENGINE = MergeTree ORDER BY S_SUPPKEY;


clickhouse-client --query "INSERT INTO customer FORMAT CSV" < customer.tbl
clickhouse-client --query "INSERT INTO part FORMAT CSV" < part.tbl
clickhouse-client --query "INSERT INTO supplier FORMAT CSV" < supplier.tbl
clickhouse-client --query "INSERT INTO lineorder FORMAT CSV" < lineorder.tbl
```
